org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/IMaven.java

```diff
@@ -148,11 +149,42 @@ MavenExecutionPlan calculateExecutionPlan(MavenProject project, List<String> goa
   MojoExecution setupMojoExecution(MavenProject project, MojoExecution execution, IProgressMonitor monitor)
       throws CoreException;
 
+  /**
+   * Instances should not be cached
+   *
+   * @since 2.1
+   */
+  public interface ConfigurationElement {
+    boolean exists();
```

Comment:
This seems weird, I would rather throw some exception when invalid names are passed. The top level element should always be valid (even if no explicit configuration is given in the underlying effective pom).

Reply:
I agree that the top-level element should always be valid and if actually absent should be treated as empty.
But since (some) elements can be present or absent, it is also a valid workflow to check if a certain element exists and only handle it if it exists. Therefore I think throwing an exception in that case is not so convenient and wanted ConfigurationElement to be similar to Optional.